elbepack/initvmaction.py

```diff
@@ -148,6 +152,54 @@ def initvm_state(self):
         return self.initvm.info()[0]
 
 
+def is_qemu_running():
+    """
+    Test if QEMU vm is running.
+
+    This function is used in QEMU mode to test if the initvm is running.
+    It's using lsof to see if a process on the current machine has bound to the
+    soap port. If such a process is found, it's expected to be the initvm.
+    """
+
+    # test is lsof command is available, and fail with error exit code if not.
+    try:
+        system('which lsof > /dev/null 2>&1')
```

Comment:
Instead of relying on multiple external commands this should use the `socket` module.

Reply:
After aligning is_qemu_running and test_soap_communication, it's now using elbe control list_projects.